api/v1alpha4/azurecluster_default_test.go

```diff
@@ -803,6 +803,253 @@ func TestNodeOutboundLBDefaults(t *testing.T) {
 				Spec: AzureClusterSpec{
 					NetworkSpec: NetworkSpec{
 						APIServerLB: LoadBalancerSpec{Type: Public},
+						Subnets: Subnets{
```

Comment:
@CecileRobertMichon can you sanity check these new scenarios, and that this is a complete representation of the cases we want to test for the outboundLB convenience func `setNodeOutboundLBDefaults`?

This is the new "default case", where we have a single control plane subnet and a single node pool without any NAT Gateway configuration. We should expect to get a default `NodeOutboundLB` resource injected into the cluster object.

Reply:
these scenarios look great, thanks for adding them